### ISSUE TITLE
Combine pushUpdate and pushMessage

### ISSUE THREAD

**Issue by geoknee** [statechannels/statechannels#3212]
# Why: Problem description :confused:
We have two very similar API methods in the server wallet: 
- `pushMessage` and
- `pushUpdate`

The former triggers objectives to be cranked. The latter throws errors unless the message is a single update for a running channel, and[ does not trigger any objectives to be cranked. ](https://github.com/statechannels/statechannels/pull/3195/commits/640fb2130f47581f8069b897377172d040c6de6d#diff-93ef0bace6dd5a6424f1f5ba77514bc2788c66488a71ad89b424b3761491dbe9R790-R801). The latter is therefore an optimised version of the former, presumably for the purpose of shortening the critical path of the wallet. 

However,`pushUpdate` is not in fact used in any consuming code at present. It would probably be used [here ](https://github.com/statechannels/graph-payments/blame/main/packages/payments/src/payment-manager.ts#L127).

# What: Describe the solution you'd like :sparkles:  

Deprecate `pushUpdate` in favour of just `pushMessage` , but make the implementation of `pushMessage` smart enough to avoid unnecessary cranking. 

Currently, `takeActions` is called during `pushUpdate`. Following the stack down, it does seem like we are reasonably careful to not crank objectives unnecessarily. We should have some tests to verify this. 

# Describe alternatives you've considered :thinking:
Do nothing, maintain two API calls. Continue to be confused about which one to use when.

**Comment by andrewgordstewart** [comment]
> Do nothing, maintain two API calls. Continue to be confused about which one to use when.

Did you consider the following alternative?

> Maintain two API calls. Ask the author to explain in the code which one should be used, when.

**Comment by geoknee** [comment]
Yeah that's a better alternative than the one I put :-P. I will update the description.
